egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,470 @@
+    speech_rir = PickItemWithProbability(room.rir_list)
+    if random.random() < speech_rvb_probability:
+        # pick the RIR to reverberate the speech
+        reverberate_opts += "--impulse-response={0} ".format(speech_rir.rir_file_location)
+
+    rir_iso_noise_list = FilterIsotropicNoiseList(iso_noise_list, speech_rir.rir_id)
+    # Add the corresponding isotropic noise associated with the selected RIR
+    if len(rir_iso_noise_list) > 0 and random.random() < isotropic_noise_addition_probability:
+        isotropic_noise = PickItemWithProbability(rir_iso_noise_list)
+        # extend the isotropic noise to the length of the speech waveform
+        noise_addition_descriptor['noise_io'].append("wav-reverberate --duration={1} {0} - |".format(isotropic_noise.noise_file_location, speech_dur))
+        noise_addition_descriptor['start_times'].append(0)
+        noise_addition_descriptor['snrs'].append(background_snrs.next())
+
+    noise_addition_descriptor = AddPointSourceNoise(noise_addition_descriptor,  # descriptor to store the information of the noise added
+                                                    room,  # the room selected
+                                                    point_noise_list, # the point source noise list
+                                                    pointsource_noise_addition_probability, # Probability of adding point-source noises
+                                                    foreground_snrs, # the SNR for adding the foreground noises
+                                                    background_snrs, # the SNR for adding the background noises
+                                                    speech_dur,  # duration of the recording
+                                                    max_noises_recording  # Maximum number of point-source noises that can be added
+                                                    )
+
+    assert len(noise_addition_descriptor['noise_io']) == len(noise_addition_descriptor['start_times'])
+    assert len(noise_addition_descriptor['noise_io']) == len(noise_addition_descriptor['snrs'])
+    if len(noise_addition_descriptor['noise_io']) > 0:
+        reverberate_opts += "--additive-signals='{0}' ".format(','.join(noise_addition_descriptor['noise_io']))
+        reverberate_opts += "--start-times='{0}' ".format(','.join(map(lambda x:str(x), noise_addition_descriptor['start_times'])))
+        reverberate_opts += "--snrs='{0}' ".format(','.join(map(lambda x:str(x), noise_addition_descriptor['snrs'])))
+
+    return reverberate_opts
+
+
+# This is the main function to generate pipeline command for the corruption
+# The generic command of wav-reverberate will be like:
+# wav-reverberate --duration=t --impulse-response=rir.wav 
+# --additive-signals='noise1.wav,noise2.wav' --snrs='snr1,snr2' --start-times='s1,s2' input.wav output.wav
+def CorruptWav(wav_scp,  # the dictionary of which elements are the IO of the speech recordings
```

Comment:
The standard term for this is 'rxfilename'



On Tue, Jul 19, 2016 at 2:03 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/wsj/s5/steps/data/reverberate_data_dir.py
> <https://github.com/kaldi-asr/kaldi/pull/706#discussion_r71422914>:
>
> > +
> > +    assert len(noise_addition_descriptor['noise_io']) == len(noise_addition_descriptor['start_times'])
> > +    assert len(noise_addition_descriptor['noise_io']) == len(noise_addition_descriptor['snrs'])
> > +    if len(noise_addition_descriptor['noise_io']) > 0:
> > +        reverberate_opts += "--additive-signals='{0}' ".format(','.join(noise_addition_descriptor['noise_io']))
> > +        reverberate_opts += "--start-times='{0}' ".format(','.join(map(lambda x:str(x), noise_addition_descriptor['start_times'])))
> > +        reverberate_opts += "--snrs='{0}' ".format(','.join(map(lambda x:str(x), noise_addition_descriptor['snrs'])))
> > +
> > +    return reverberate_opts
> > +
> > +
> > +# This is the main function to generate pipeline command for the corruption
> > +# The generic command of wav-reverberate will be like:
> > +# wav-reverberate --duration=t --impulse-response=rir.wav
> > +# --additive-signals='noise1.wav,noise2.wav' --snrs='snr1,snr2' --start-times='s1,s2' input.wav output.wav
> > +def CorruptWav(wav_scp,  # the dictionary of which elements are the IO of the speech recordings
>
> #the dictionary of which elements are the IO of the speech recordings
> -->
> `#a dictionary whose values are the Kaldi-IO strings of the speech
> recordings
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/706/files/d335c718e3b1849afe40a61fc436c0e098c6a36e#r71422914>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu6n4Fgl9AK7KhWxsSSfPojQ_kJq-ks5qXTufgaJpZM4IJM2C>
> .
>